src/main.ts

```diff
@@ -34,14 +36,31 @@ export async function run(): Promise<void> {
       )
       execSync(`docker context use target`, { stdio: [] })
     } catch (error: unknown) {
-      core.setFailed(`Failed to initialise context: ${error}`)
+      throw new Error(`Failed to create docker context: ${error}`)
+    }
+
+    const parsedSecrets = parseSecrets(secrets)
+    if (parsedSecrets !== undefined) {
+      for (const secret of parsedSecrets) {
+        core.info(`Creating secret ${secret.name} ...`)
+        try {
+          execSync(
+            `docker secret inspect ${secret.name} || echo ${secret.value} | docker secret create ${secret.name} -`,
```

Comment:
Would it make sense to use `spawnSync("docker", ["secret", ...])` here instead of `exec`? It's not about security here, but I think it will lead to better error messages if you e.g. have a space or some invalid character in the `secret.name`

Reply:
`execFile` might also be an option